lib/error_message.ex

```diff
@@ -8,14 +8,59 @@ defmodule ErrorMessage do
   @enforce_keys [:code, :message]
   defstruct [:code, :message, :details]
 
-  @type t ::
-          %ErrorMessage{code: atom, message: String.t()}
-          | %ErrorMessage{code: atom, message: String.t(), details: any}
-
-  @type t_map ::
-          %{code: atom, message: String.t()}
-          | %{code: atom, message: String.t(), details: any}
-
+  @type code ::
```

Comment:
Added type code to replace atom as there is a subset of codes that are valid and this would help a person if they accidentally misspelled one of them in a pattern match.